include/inset_state.h

```diff
@@ -165,27 +215,57 @@ class InsetState
   bool target_area_is_missing(const std::string &) const;
   double total_inset_area() const;
   double total_target_area() const;
+  Polygon transform_to_albers_coor(Polygon edge_points);
   std::array<Point, 3> transformed_triangle(
     const std::array<Point, 3> &,
     bool = false) const;
 
   // Apply given function to all points
   void transform_points(const std::function<Point(Point)> &, bool = false);
   std::array<Point, 3> untransformed_triangle(Point, bool = false) const;
+  void trim_grid_heatmap(cairo_t *cr, double padding);
+  std::array<Point, 3> untransformed_triangle(const Point, bool = false);
 
   // Cairo functions
-  void write_cairo_map(const std::string &, bool);
-  void write_cairo_polygons_to_png(const std::string &, bool, bool, bool);
-  void write_cairo_polygons_to_ps(const std::string &, bool, bool, bool);
+  void write_cairo_map(
+    const std::string &,
+    bool,
+    std::unordered_map<Point, Point> = std::unordered_map<Point, Point>());
+  void write_cairo_polygons_to_svg(
+    const std::string &,
+    bool,
+    bool,
+    bool,
+    std::unordered_map<Point, Point> &);
+
+  void write_delaunay_triangles(const std::string &);
+  void write_grid_heatmap_data(const std::string filename);
 
   // Functions to write map to eps
+  // void write_density_to_eps(const std::string, const double *);
+  void write_grid_heatmap_image(
+    const std::string filename,
+    const bool plot_equal_area_map,
+    const bool crop_polygons);
+  void write_grids_on_surface(cairo_t *cr);
+  void write_grid_colors_on_surface(
+    cairo_t *cr,
+    bool plot_equal_area_map,
+    bool crop_polygons);
+  void write_polygons_on_surface(
+    cairo_t *cr,
+    const bool fill_polygons,
+    const bool colors,
+    const bool plot_equal_area_map);
+  void write_labels_on_surface(cairo_t *cr);
+  void write_density_image(
+    const std::string filename,
+    const double *density,
+    const bool plot_pycnophylactic);
+  void write_intersections_image(unsigned int res);
+  void write_legend_on_surface(cairo_t *cr, bool equal_area_map);
```

Comment:
This is a utility function to be called manually in the main plotting function for customizing figures. Since this feature is not very important, a flag was not created.